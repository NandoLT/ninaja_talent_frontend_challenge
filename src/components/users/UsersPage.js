import React, { useEffect, useState } from 'react';
import Layout from "../commons/Layout";
import { getUsers } from '../../services/api/users/usersData';
import UsersList from './UsersList';
import Loader from '../commons/Loader';

const EmptyList = () => {
    return (
        <div style={{ textAlign: 'center' }}>
            <p>Users List Empty !!!</p>
        </div>
    )
}

const  UsersPage = () => {

    const initialState = [];
    const [users, setUsers] = useState(initialState);
    const [error, setError] = useState(false);
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        getUsers()
        .then(setUsers)
        .catch( error => {
            console.log(error);
            setError(error);
        })
        .finally( () => {
            setIsLoading(!isLoading);
            }
            );
    }, []);

    return (
        <Layout >
            {error ?
                <h2> Something go wrong, please try it again later </h2> 
                :
                isLoading ? 
                    <Loader />
                    :
                        users.length ?
                            <UsersList users={ users } /> :
                            <EmptyList />
            }
        </Layout>
    )
}

export default UsersPage;

        