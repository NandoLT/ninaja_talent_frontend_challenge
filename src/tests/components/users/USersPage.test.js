import { shallow } from 'enzyme';
import React from 'react';
import UsersPage from '../../../components/users/UsersPage';

describe('Tests in <UsersPage />', () => {

    test('Should display correctly', () => {

        const wrapper = shallow(<UsersPage />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should display a list of users', () => {

        const wrapper = shallow(<UsersPage />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('UsersList').length).toBe(1);
    });
})