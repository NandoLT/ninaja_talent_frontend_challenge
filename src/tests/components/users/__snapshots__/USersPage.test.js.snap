// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <UsersPage /> Should display correctly 1`] = `
<Layout>
  <EmptyList />
</Layout>
`;

exports[`Tests in <UsersPage /> should display a list of users 1`] = `
<Layout>
  <EmptyList />
</Layout>
`;
